### ISSUE TITLE
Develop an executable python script to create YAML files for JEDI executables

### ISSUE THREAD

**Issue by CoryMartin-NOAA** [NOAA-EMC/GDASApp#33]
We need a generic, executable python tool that can create YAML from environment variables and/or an input YAML file for use in JEDI executables.

Something like this:
`genJEDIyaml --yaml $PSLOT.yaml`

Things can either be set from environment variables:
`export CDATE=2022032818`
or in that YAML:
`CDATE: 2022032818`

I suspect the best option is to read env first, and then read YAML (so the YAML can overwrite if need be)

Additionally, everything called in the main script should also be callable from a python function and passed a `dict` rather than read a YAML.

Thoughts on features, design, etc. @RussTreadon-NOAA @guillaumevernieres?

**Comment by CoryMartin-NOAA** [comment]
Starting work on https://github.com/NOAA-EMC/GDASApp/tree/feature/yamlgen

Current idea would be:
`genYAML --yaml myexp.yaml --output 3dvar.yaml`
Is `yaml` too generic? Should I say `input`? This YAML would be optional and just overwrite certain ENV vars.

Edit: changed yaml to config:
`genYAML --config myexp.yaml --output 3dvar.yaml`

**Comment by guillaumevernieres** [comment]
> Starting work on https://github.com/NOAA-EMC/GDASApp/tree/feature/yamlgen
> 
> Current idea would be: `genYAML --yaml myexp.yaml --output 3dvar.yaml` Is `yaml` too generic? Should I say `input`? This YAML would be optional and just overwrite certain ENV vars.
> 
> Edit: changed yaml to config: `genYAML --config myexp.yaml --output 3dvar.yaml`

:+1:  on ``--config`` , more meaningful that ``--yaml`. I like ``--input`` as well.
is ``myexp.yaml`` a template?

**Comment by guillaumevernieres** [comment]
Or given, a set of necessary keys and a type of application we want to use (jedi or otherwise), generate the yaml for that?
``genYAML --config myexp.yaml --type 3dvar --output 3dvar.yaml``

**Comment by CoryMartin-NOAA** [comment]
after chatting with @aerorahul, we think it's best to stick with a template approach for now.

`genYAML --config myexp.yaml`

myexp.yaml:
```
template: /path/to/3dvar.yaml
output: /path/to/my/exp/3dvar_out.yaml
config:
  user_var1: foo
  user_var2: bar
```

Anything in config can replace placeholders in the templates.

**Comment by CoryMartin-NOAA** [comment]
Closed through #35
